ably/options.go

```diff
@@ -337,45 +340,84 @@ func (opts *ClientOptions) disconnectedRetryTimeout() time.Duration {
 	return defaultOptions.DisconnectedRetryTimeout
 }
 
-func (opts *ClientOptions) getHost() (restHost string, realtimeHost string) {
-	host := func(host string, defaultHost string) string {
-		if host == "" {
-			host = defaultHost
-		}
-		if host == defaultHost && !opts.isProductionEnvironment(){
-				host = opts.Environment + "-" + host
+func (opts *ClientOptions) getRestHost() (string) {
+	host := opts.RestHost
+	defaultHost := defaultOptions.RestHost
+	if empty(host) {
+		host = defaultHost
+	}
+	if host == defaultHost && !opts.isProductionEnvironment(){
+			host = opts.Environment + "-" + host
+	}
+	return host
+}
+
+func (opts *ClientOptions) getRealtimeHost() (string) {
+	host := opts.RealtimeHost
+	defaultHost := defaultOptions.RealtimeHost
+	if empty(host) {
+		if !empty(opts.RestHost) {
+			logger := opts.Logger.Sugar()
+			logger.Warnf(`restHost is set to %s but realtimeHost is not set so setting realtimeHost to %s too. 
```

Comment:
I'm not sure we should be logging this, but even if we do, shouldn't we log this if it's the other way around as well? Namely, that `realtimeHost` is set but `restHost` isn't.

Reply:
I believe that's how it's implemented in other libraries as per spec.

Reply:
Sorry, I missed that this actually overrides the default realtime host.